### PR TITLE
PTP: Print parts of the common PTP header

### PR DESCRIPTION
**Pull request by lunn**
Precession Time Protocol allows network devices to synchronize their
clocks. Each PTP packet has a common header, and then message type
specific extensions. Print out some of the important fields in the
common header.

PTP can be carried directly in Ethernet frames, or UDP
encapsulated. Add support for both.

Signed-off-by: Andrew Lunn <andrew@lunn.ch>

### CODE REVIEW

**Comment by lunn** [comment]
It would be nice to add to CONTRIBUTING a list of 'Makefile' like files where new files need to be added in order that they get built in different build systems...

File: print-ptp.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	uint8_t messageType;
+	uint16_t flags;
+	uint16_t seq;
+	uint64_t clockId;
+	uint16_t srcId;
+
+	if (length < PTP_COMMON_HEADER_LEN)
+		goto trunc;
+
+	messageType = 	EXTRACT_U_1(&bp[0]) & 0xf;
```
Comment (guyharris) [inline print-ptp.c]: There should be ND_TCHECK_n() calls before fetching; `length` is the on-the-wire length, not the captured length.

Comment (lunn) [reply print-ptp.c]: Done

Comment (lunn) [reply print-ptp.c]: Trivial merge conflict resolved

**Comment by lunn** [comment]
Anything i can do to help get this merged?

**Comment by infrastation** [comment]
This decoder should take into account the `versionPTP` field (4 lower bits of octet 1 in the header) and only decode the version it implements (from v2 packet diagrams I infer this code stands for v1). Other than that, it looks very straightforward.

**Comment by infrastation** [comment]
To improve `CONTRIBUTING` as you suggested above, feel free to suggest particular changes through a pull request.

**Comment by fxlb** [comment]
PR #721 is also about PTP.

**Comment by mcr** [comment]
> PR #721 is also about PTP.

I think #721 is probably a better base, do you agree?  Would the two posters like to collaborate?

**Comment by fxlb** [comment]
The implementation was done via PR #721, closing.

**Pull request closed without merging**